src/object/reader.rs

```diff
@@ -186,6 +190,56 @@ impl AsyncSeek for ObjectReader {
     }
 }
 
+impl tokio::io::AsyncRead for ObjectReader {
+    fn poll_read(
+        mut self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+        buf: &mut tokio::io::ReadBuf<'_>,
+    ) -> Poll<io::Result<()>> {
+        let b = buf.initialize_unfilled();
+        let n = ready!(futures::AsyncRead::poll_read(self.as_mut(), cx, b))?;
+        unsafe {
+            buf.assume_init(n);
+        }
+        buf.advance(n);
+        Poll::Ready(Ok(()))
+    }
+}
+
+impl tokio::io::AsyncSeek for ObjectReader {
+    fn start_seek(self: Pin<&mut Self>, pos: io::SeekFrom) -> io::Result<()> {
+        let this = self.get_mut();
+        this.seek_state = SeekState::Start(pos);
```

Comment:
Maybe we need to do a check here. If users tried to call `start_seek` while there is a seek on-going, we need to return an error instead.

> This function can return [io::ErrorKind::Other](https://doc.rust-lang.org/nightly/std/io/error/enum.ErrorKind.html#variant.Other) in case there is another seek in progress.